utils/data-migration/migrate-acbs-utilisation-data-to-tfm/index.ts

```diff
@@ -0,0 +1,61 @@
+import { SqlDbDataSource } from '@ukef/dtfs2-common/sql-db-connection';
+import { FacilityUtilisationDataEntity, ReportPeriodPartialEntity } from '@ukef/dtfs2-common';
+import { DataSource } from 'typeorm';
+import { existsSync, readFileSync } from 'fs';
+import { resolve } from 'path';
+
+const JSON_FILE_PATH = resolve(__dirname, 'acbs-utilisation-data.json');
+
+const REPORT_PERIOD: ReportPeriodPartialEntity = {
+  start: { month: 1, year: 2020 },
+  end: { month: 3, year: 2020 },
+};
+
+type UtilisationDataEntry = {
+  facilityId: string;
+  utilisation: number;
+  fixedFeePremium: number;
+};
+
+const validateReportPeriod = (reportPeriod: ReportPeriodPartialEntity): void => {
+  const { start, end } = reportPeriod;
+  if (start.year > end.year || (start.year === end.year && start.month > end.month)) {
+    throw new Error('Report period end date cannot be before start date');
+  }
+};
+
+const insertFacilityUtilisationEntry = async (dataSource: DataSource, { facilityId, utilisation, fixedFeePremium }: UtilisationDataEntry) => {
+  const facilityUtilisationData = new FacilityUtilisationDataEntity();
+  facilityUtilisationData.id = facilityId;
+  facilityUtilisationData.reportPeriod = REPORT_PERIOD;
+  facilityUtilisationData.utilisation = utilisation;
+  facilityUtilisationData.fixedFee = fixedFeePremium;
+  facilityUtilisationData.updateLastUpdatedBy({ platform: 'SYSTEM' });
+
+  await dataSource.manager.save(FacilityUtilisationDataEntity, facilityUtilisationData);
+};
+
+const run = async () => {
+  validateReportPeriod(REPORT_PERIOD);
+
+  const dataSource = await SqlDbDataSource.initialize();
+
+  try {
+    if (!existsSync(JSON_FILE_PATH)) {
+      throw new Error(`Migration JSON file not found at path: ${JSON_FILE_PATH}`);
+    }
+    const jsonData = readFileSync(JSON_FILE_PATH, 'utf8');
+
+    const utilisationData = JSON.parse(jsonData) as UtilisationDataEntry[];
+
+    await Promise.all(utilisationData.map((entry) => insertFacilityUtilisationEntry(dataSource, entry)));
```

Comment:
I imagine the numbers of entries will only be in the low 200s - could you test this script against your local DB with a file with 250 entries please?
if it works then im more than happy with this and no changes needed - in future I'd probably map to the entities to get the full list of entities then call a single save to the database with a batch size of 100 (so 100 get saved at a time)

Sorry missed this on first review!

Reply:
Ran with 250 entries in the JSON, didn't get any errors. Happy to implement the batching if you'd prefer

![Screenshot 2024-08-22 at 16 20 04](https://github.com/user-attachments/assets/87667e4e-cc14-4969-a6cf-24f98a7a6028)

Reply:
Batching shouldn't be required here as we are running one insert per document :)